src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -519,21 +512,29 @@ public byte[] Http2AltSvcOriginUri
             Socket? socket = null;
             SslStream? sslStream = null;
             TransportContext? transportContext = null;
+            Http2Connection[]? lastHttp2Connections = _http2Connections;
 
             // Serialize creation attempt
             await _http2ConnectionCreateLock.WaitAsync(cancellationToken).ConfigureAwait(false);
             try
             {
-                if (_http2Connection != null)
+                Http2Connection[]? currentHttp2Connections = _http2Connections;
+                if (EnableMultipleHttp2Connections)
                 {
-                    // Someone beat us to it
-
-                    if (NetEventSource.Log.IsEnabled())
+                    if (_http2Connections != null)
```

Comment:
GetExistingHttp2Connection does this check anyway, right? So this test is unnecessary.